src/app/redux/UserSaga.js

```diff
@@ -407,13 +467,32 @@ function* promptTosAcceptance(username) {
 
 function* getFeatureFlags(username, posting_private) {
     try {
-        const flags = yield call(
-            [api, api.signedCallAsync],
-            'conveyor.get_feature_flags',
-            { account: username },
-            username,
-            posting_private
-        );
+        let flags;
+        if (hasCompatibleKeychain()) {
+            flags = yield new Promise((resolve, reject) => {
+                window.steem_keychain.requestSignedCall(
```

Comment:
Just tested this in my browser console.

This is a significant change in UX as it will trigger a popup and interrupt loading instead of making a background call.
Also currently the keychain extension fails to use higher order keys, so if I have "active" key set, this function will fail because it requires "posting" key, with no way to recover it.

Reply:
Ah, the feature flag fetching is already triggered by a background call earlier (a fork), so I don't believe it is the case that it interrupts loading.

Hmm... as for the higher order key, could be a keychain behavior to allow for active key to sign instead of a posting. But I'm not sure. Would that work?